prov/verbs/src/windows/verbs_nd_rdma.c

```diff
@@ -168,15 +171,64 @@ int rdma_destroy_id(struct rdma_cm_id *id)
 	}
 
 	id_nd = container_of(id, struct nd_cm_id, id);
+	ofi_mutex_lock(&id_nd->connect_event.base.lock);
+	ofi_mutex_lock(&id_nd->peer_event.base.lock);
+	ofi_mutex_lock(&id_nd->listen_event.base.lock);
+	if (id_nd->connect_event.base.cb_pending ||
+	    id_nd->peer_event.base.cb_pending)
+		id_nd->connector->lpVtbl->CancelOverlappedRequests(
+			id_nd->connector);
+
+	if (id_nd->listen_event.base.cb_pending)
+		id_nd->listener->lpVtbl->CancelOverlappedRequests(
+			id_nd->listener);
+
 	while (compare = ofi_atomic_get32(&id_nd->cm_events_pending)) {
 		WaitOnAddress(&id_nd->cm_events_pending, &compare,
```

Comment:
Waiting on an event for an infinite time while holding 3 mutexes gives the impression of bad form.  Why do we need to hold these locks?